src/components/Checkbox.js

```diff
@@ -1,65 +1,30 @@
-import React, { useCallback } from 'react'
-import propTypes from 'prop-types'
-import { Field, Checkbox as CoreCheckbox, Help } from '@dhis2/ui-core'
+import React from 'react'
+import propTypes from '@dhis2/prop-types'
+import { CheckboxField } from '@dhis2/ui-core'
 
 import { FieldAdapter, adapterComponentProps } from './FieldAdapter.js'
+import { useToggleChangeHandler } from './shared/hooks.js'
 
-const Checkbox = ({
-    label,
-    required,
-    disabled,
-    name,
-    value,
-    checkedValue,
-    onChange,
-    error,
-    warning,
-    valid,
-    helpText,
-    errorText,
-}) => (
-    <Field>
-        <CoreCheckbox
-            required={required}
-            name={name}
-            disabled={disabled}
-            value={checkedValue}
-            label={label}
+const CheckboxComponent = ({ onChange, value, checkedValue, ...rest }) => {
+    const handleChange = useToggleChangeHandler(onChange)
+
+    return (
+        <CheckboxField
+            {...rest}
             checked={!!value}
-            onChange={onChange}
-            error={error}
-            warning={warning}
-            valid={valid}
+            value={checkedValue}
```

Comment:
I think I'd prefer to have to specify `type="checkbox" value="foo"` than `checkedValue="foo"`.
I understand that it's a nuisance having to add that addition prop when the component already indicated that it's a checkbox, but I'm still preferring what I just stated for the following reasons:

1. we've decided to create this library in a way so we don't alter the way someone has to work with final form. And although this is a nice, little addition by `ui-forms` to not having to add the type to the `Field` component, it's a deviation from what we are trying to do
1. `checkedValue` is a bit odd, if we really want to keep this way of passing in the value, I'd prefer `actualValue` or `fieldValue` as the word `checked` is already something you have to keep in mind for fields that are "checked" (which refers back to 1.: I have to remember that these are special anyways, why not sticking to the standard, it'd feel much more natural to pass in the value through a `value` prop).

Reply:
After having worked this a bit, I think it's ok like this actually